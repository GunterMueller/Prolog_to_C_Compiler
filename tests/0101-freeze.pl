main :-
	freeze(X, (display('X='), display(X), nl)),
	foo(X),
	bar,
	X = 1,
	bar,
	done.

foo(Y).

done :- display('done\n').

bar :- display(bar), nl.

