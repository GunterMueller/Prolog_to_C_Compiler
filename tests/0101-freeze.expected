bar
X=1
bar
done
